components/org.wso2.carbon.identity.tag.manager.core/src/main/java/org/wso2/carbon/identity/tag/manager/core/dao/TagManagementDAOImpl.java

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.tag.manager.core.dao;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.tag.manager.core.constant.Constants;
+import org.wso2.carbon.identity.tag.manager.exception.TagServiceException;
+import org.wso2.carbon.identity.tag.manager.model.ErrorMessage;
+import org.wso2.carbon.identity.tag.manager.model.Tag;
+import org.wso2.carbon.identity.tag.manager.model.TagAssociationsResult;
+import org.wso2.carbon.identity.tag.manager.model.TagListResult;
+import org.wso2.carbon.registry.core.utils.UUIDGenerator;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tag management service DAO implementation class.
+ */
+public class TagManagementDAOImpl implements TagManagementDAO {
+
+    @Override
+    public Tag loadTag(String tagUuid, String tenantUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        ResultSet resultSet;
+        Tag tag = null;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_IDN_TAG_BY_TAG_UUID_SQL);
```

Comment:
In previous code reviews, we have decided to move the PreparedStatement to the resource block. Is it possible move? if it is possible lets fix in other places as well.

Reply:
fixed